Library/Formula/infer.rb

```diff
@@ -0,0 +1,116 @@
+class Infer < Formula
+  desc "Static analyzer for Java, C and Objective-C"
+  homepage "http://fbinfer.com/"
+  url "https://github.com/facebook/infer/releases/download/v0.5.0/infer-osx-v0.5.0.tar.xz"
+  sha256 "6a8547ac0b75a5e2bbeccae2169e39f753a60adbcacb6c94599fd31343a71ce7"
+
+  option "without-clang", "Build without C/Objective-C analyzer"
+  option "without-java", "Build without Java analyzer"
+
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
+  depends_on "libtool" => :build
+  depends_on "opam" => :build
+
+  def install
+    if build.without?("clang") && build.without?("java")
+      odie "infer: --without-clang and --without-java are mutually exclusive"
+    end
+    opamroot = buildpath/"build"
+    opamroot.mkpath
+    ENV["OPAMROOT"] = opamroot
+    ENV["OPAMYES"] = "1"
+
+    system "opam", "init", "--no-setup"
+    system "opam", "update"
+
+    system "opam", "install", "ocamlfind"
+    system "opam", "install", "sawja>=1.5.1"
+    system "opam", "install", "atdgen>=1.6.0"
+    system "opam", "install", "extlib>=1.5.4"
+
+    target_platform = if build.without?("clang")
+      "java"
+    elsif build.without?("java")
+      "clang"
+    else
+      "all"
+    end
+    system "./build-infer.sh", target_platform, "--yes"
+
+    libexec.install "facebook-clang-plugins"
+    libexec.install "infer/"
+
+    bin.install_symlink libexec/"infer/bin/infer"
+  end
+
+  test do
+    if build.with?("clang")
```

Comment:
I'd really rather we didn't include anything that isn't the default. Testing non-default options opens a floodgate of testing possibilities that we're not likely to be able to cram back into the jar.

We don't build non-default options on CI so not building them to know whether they still work or not and then allowing users to test those non-default options seems like a recipe for user confusion and maintainer frustration.